lib/ansible/modules/cloud/google/gcp_compute_firewall.py

```diff
@@ -0,0 +1,477 @@
+
+DOCUMENTATION = '''
+---
+module: gcp_compute_firewall
+description:
+    - Each network has its own firewall controlling access to and from the instances.
+    - All traffic to instances, even from other instances, is blocked by the firewall
+      unless firewall rules are created to allow it.
+    - The default network has automatically created firewall rules that are shown in default
+      firewall rules. No manually created network has automatically created firewall rules
+      except for a default "allow" rule for outgoing traffic and a default "deny" for
+      incoming traffic. For all networks except the default network, you must create any
+      firewall rules you need.
+short_description: Creates a GCP Firewall
+version_added: 2.6
+author: Google Inc. (@googlecloudplatform)
+requirements:
+    - python >= 2.6
+    - requests >= 2.18.4
+    - google-auth >= 1.3.0
+options:
+    state:
+        description:
+            - Whether the given object should exist in GCP
+        choices: ['present', 'absent']
+        default: 'present'
+    allowed:
+        description:
+            - The list of ALLOW rules specified by this firewall. Each rule specifies a protocol
+              and port-range tuple that describes a permitted connection.
+        required: false
+        suboptions:
+            ip_protocol:
+                description:
+                    - The IP protocol to which this rule applies. The protocol type is required when creating
+                      a firewall rule. This value can either be one of the following well known protocol
+                      strings (tcp, udp, icmp, esp, ah, sctp), or the IP protocol number.
+                required: true
+            ports:
+                description:
+                    - An optional list of ports to which this rule applies. This field is only applicable
+                      for UDP or TCP protocol. Each entry must be either an integer or a range. If not
+                      specified, this rule applies to connections through any port.
+                    - 'Example inputs include: ["22"], ["80","443"], and ["12345-12349"].'
+                required: false
+    description:
+        description:
+            - An optional description of this resource. Provide this property when you create
+              the resource.
+        required: false
+    name:
+        description:
+            - Name of the resource. Provided by the client when the resource is created. The name
+              must be 1-63 characters long, and comply with RFC1035. Specifically, the name must
+              be 1-63 characters long and match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`
+              which means the first character must be a lowercase letter, and all following characters
+              must be a dash, lowercase letter, or digit, except the last character, which cannot
+              be a dash.
+        required: false
+    network:
```

Comment:
This isn't used in the examples. Is it possible to do this by name instead of URL? It seems a little complex to have to provide a full URL to use non-default networks.